src/main/java/com/orbitz/consul/SessionClient.java

```diff
@@ -8,176 +10,202 @@
+            target = webTarget.queryParam("dc", dc);
+        }
+
+        SessionCreatedResponse session = target.path("create").request().put(Entity.entity(value,
+                MediaType.APPLICATION_JSON_TYPE), SESSION_CREATED_RESPONSE_TYPE);
+
+        return session;
+    }
+
+    public Optional<SessionInfo> renewSession(final String sessionId) {
+        return renewSession(null, sessionId);
+    }
+
+    /**
+     * Renews a session.
+     *
+     * @param dc        The datacenter.
+     * @param sessionId The session ID to renew.
+     * @return The {@link SessionInfo} object for the renewed session.
+     */
+    public Optional<SessionInfo> renewSession(final String dc, final String sessionId) {
+        WebTarget target = webTarget;
+
+        if (dc != null) {
+            target = webTarget.queryParam("dc", dc);
+        }
+
+        List<SessionInfo> sessionInfo = target.path("renew").path(sessionId).request().put(Entity.entity("{}",
+                MediaType.APPLICATION_JSON_TYPE), SESSION_INFO_LIST_TYPE);
+
+        return sessionInfo != null && sessionInfo.isEmpty() ? Optional.<SessionInfo>absent() :
+                Optional.of(sessionInfo.get(0));
+    }
+
+    /**
+     * Destroys a session.
+     * 
+     * PUT /v1/session/destroy/{sessionId}
+     *
+     * @param sessionId The session ID to destroy.
+     */
+    public void destroySession(final String sessionId) {
+        destroySession(sessionId, null);
+    }
+
+    /**
+     * Destroys a session.
+     * 
+     * PUT /v1/session/destroy/{sessionId}
+     *
+     * @param sessionId The session ID to destroy.
+     * @param dc        The data center.
+     */
+    public void destroySession(final String sessionId, final String dc) {
+        WebTarget target = webTarget;
+
+        if (dc != null) {
+            target = webTarget.queryParam("dc", dc);
+        }
+
+        target.path("destroy").path(sessionId).request().put(Entity.entity("",
+                MediaType.TEXT_PLAIN_TYPE));
+    }
+
+    /**
+     * Retrieves session info.
+     * 
+     * GET /v1/session/info/{sessionId}
+     *
+     * @param sessionId
+     * @return {@link SessionInfo}.
+     */
+    public Optional<SessionInfo> getSessionInfo(final String sessionId) {
+        return getSessionInfo(sessionId, null);
+    }
+
+    /**
+     * Retrieves session info.
+     * 
+     * GET /v1/session/info/{sessionId}
+     *
+     * @param sessionId
+     * @param dc        Data center
+     * @return {@link SessionInfo}.
+     */
+    public Optional<SessionInfo> getSessionInfo(final String sessionId, final String dc) {
+        WebTarget target = webTarget;
+
+        if (dc != null) {
+            target = target.queryParam("dc", dc);
+        }
+
+        target = addParams(target.path("info").path(sessionId), QueryOptions.BLANK);
+
+        List<SessionInfo> sessionInfo = Arrays.asList(target
+                .request().accept(MediaType.APPLICATION_JSON_TYPE).get(SessionInfo[].class));
```

Comment:
shouldn't you be using `SESSION_INFO_LIST_TYPE` here instead of going from an array to a list?